nwapp/utils/notifier.js

```diff
@@ -28,14 +36,85 @@ function playNotificationSound() {
   audio = null;
 }
 
+
+// Cache lookup
+var urlFilePathMap = new Map();
+function getFileAtUrl(url) {
+  return new Promise(function(resolve, reject) {
+    var cachedFilePath = urlFilePathMap.get(url);
+    if(cachedFilePath) {
+      resolve(cachedFilePath);
+    }
+    else {
+      setTimeout(function() {
+        reject(new Error('Took to long to request file', url));
+      }, 750);
+
+      var tempFileStream = temp.createWriteStream();
+      tempFileStream
+        .on('finish', function() {
+          var tempFilePath = tempFileStream.path;
+          // Store it on our map as a cache lookup
+          urlFilePathMap.set(url, tempFilePath);
+          resolve(tempFilePath);
+        })
+        .on('error', function(err) {
+          reject(err);
+        });
+
+      request(url)
+        .on('error', function(err) {
+          reject(err);
+        })
+        .pipe(tempFileStream);
+    }
+  });
+}
+
 module.exports = function (spec) {
```

Comment:
@trevorah Those properties are all optional. I have cleaned them up and made sure we aren't using anything when it isn't available.